test/test_init_db.py

```diff
@@ -0,0 +1,207 @@
+    mc.push_project(project_dir)
+
+    # run sync
+    dbsync_pull(mc)
+    dbsync_push(mc)
+
+    # check that new feature was added
+    cur.execute(f'SELECT COUNT(*) from {db_schema_main}."simple"')
+    assert cur.fetchone()[0] == 4
+
+
+def test_with_db_changes(mc: MerginClient, db_connection):
+    project_name = "test_db_changes_mergin"
+    project_full_name = complete_project_name(project_name)
+    project_dir = name_project_dir(project_name)
+    db_schema_main = name_db_schema_main(project_name)
+    db_schema_base = name_db_schema_base(project_name)
+
+    init_sync_from_db(mc, project_name)
+
+    cur = db_connection.cursor()
+
+    cur.execute(
+        f'INSERT INTO "{db_schema_main}"."simple" ("wkb_geometry" , "fid", "name", "rating") VALUES (\'0101000020E61000009CB92A724E60E7BFE0FDF1F774B6A53F\', 4, \'new feature\', 4);'
+    )
+    cur.execute("COMMIT")
+
+    dbsync_pull(mc)
+    dbsync_push(mc)
+
+    mc.download_project(project_full_name, project_dir)
+
+    # look at changes in in GPKG
+    gpkg_conn = sqlite3.connect(project_dir + "/" + filename_sync_gpkg())
+    gpkg_cur = gpkg_conn.cursor()
+    gpkg_cur.execute("SELECT COUNT(*) FROM simple")
+    assert gpkg_cur.fetchone()[0] == 4
+
+
+def test_missing_table(mc: MerginClient, db_connection):
+    project_name = "test_db_missing_table"
+    project_full_name = complete_project_name(project_name)
+    project_dir = name_project_dir(project_name)
+    db_schema_main = name_db_schema_main(project_name)
+    db_schema_base = name_db_schema_base(project_name)
+    sync_project_dir = name_project_sync_dir(project_name)  # used by dbsync
+
+    cleanup(
+        mc,
+        project_full_name,
+        [
+            project_dir,
+            sync_project_dir,
+        ],
+    )
+    cleanup_db(
+        db_connection,
+        db_schema_base,
+        db_schema_main,
+    )
+
+    with open(
+        path_test_data("create_base.sql"),
+        encoding="utf-8",
+    ) as file:
+        base_table_dump = file.read()
+
+    base_table_dump = base_table_dump.replace("default-schema-name", db_schema_main)
+    base_table_dump = base_table_dump.replace("default-table-name", "simple")
+
+    cur = db_connection.cursor()
+    cur.execute(base_table_dump)
+
+    # prepare a new Mergin Maps project
+    mc.create_project(
+        project_name,
+        namespace=WORKSPACE,
+    )
+
+    connection = {
+        "driver": "postgres",
+        "conn_info": DB_CONNINFO,
+        "modified": "non_existing_schema",
+        "base": db_schema_base,
+        "mergin_project": project_full_name,
+        "sync_file": filename_sync_gpkg(),
+    }
+
+    config.update(
+        {
+            "GEODIFF_EXE": GEODIFF_EXE,
+            "WORKING_DIR": sync_project_dir,
+            "MERGIN__USERNAME": API_USER,
+            "MERGIN__PASSWORD": USER_PWD,
+            "MERGIN__URL": SERVER_URL,
+            "CONNECTIONS": [connection],
+            "init_from": "db",
+        }
+    )
```

Comment:
this looks like a copy of init_sync_from_db() - to reduce code duplication I would suggest to reuse init_sync_from_db(), and make it accept any SQL file to populate the DB (just like init_sync_from_geopackage() gets source GPKG file), then add another .sql file in addition to create_base.sql that would populate a different schema...